src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length / 2; i++) {
```

Comment:
Using `figures.length / 2` directly in the loop condition is not ideal because it's calculated with each iteration. Consider storing this value in a local variable before the loop starts for better performance.